yt/data_objects/tests/test_center_squeeze.py

```diff
@@ -0,0 +1,57 @@
+from yt.testing import assert_array_equal, fake_amr_ds, fake_particle_ds, fake_random_ds
+
+
+def test_center_squeeze():
+    # tests that selected values match when supplying center arrays of different shapes
+    # to the data container.
+
+    # list of fields to populate fake datasets with
+    fldz = ("density", "velocity_x", "velocity_y", "velocity_z")
+
+    # create and test amr, random and particle data
+    check_single_ds(fake_amr_ds(fields=fldz))
+    check_single_ds(fake_random_ds(16, fields=fldz))
+    check_single_ds(fake_particle_ds(npart=100), check_morton=False)
+
+
+def check_single_ds(ds, check_morton=True):
+    # compares values for range of data containers using different center array shapes
+
+    center = ds.domain_center  # reference center array
+
+    # build some data containers
+    sp0 = ds.sphere(center, 0.25)
+    sl0 = ds.slice(0, 0.25, center=center)
+    reg0 = ds.region(center, [-0.25, -0.25, -0.25], [0.25, 0.25, 0.25])
+
+    # store morton indices of each
+    i_sp0 = None
+    i_sl0 = None
+    i_reg0 = None
+    if check_morton:
+        i_sp0 = sp0["index", "morton_index"]
+        i_sp0.sort()
+        i_sl0 = sl0["index", "morton_index"]
+        i_sl0.sort()
+        i_reg0 = reg0["index", "morton_index"]
+        i_reg0.sort()
+
+    # create new containers for different shapes of the center array
+    for test_shape in [(1, 3), (1, 1, 3)]:
+        new_center = center.reshape(test_shape)
+        sp = ds.sphere(new_center, 0.25)
+        sl = ds.slice(0, 0.25, center=new_center)
+        reg = ds.region(new_center, [-0.25, -0.25, -0.25], [0.25, 0.25, 0.25])
+
+        # compare each to the reference containers
+        for ob, ob0, i_ob0 in [(sp, sp0, i_sp0), (sl, sl0, i_sl0), (reg, reg0, i_reg0)]:
+
+            # check that selection field values match the reference
+            for fld in ds.field_list:
```

Comment:
Isn't it an overkill to check more than an a single field here ?

Reply:
yes, probably :) do you think it's also overkill to test the 3 types of test data types? i.e., it's testing `fake_amr_ds`, `fake_random_ds` and `fake_particle_ds`. I erred on the side of over testing since `data_containers` is used by virtually everything, but maybe they're not all needed?

Reply:
I think it's alright, the fact that those 3 data set types are using the same code here is an implementation detail and may not be true in the future, so it's good to test all three imo, but other reviewers might disagree :)

Reply:
I don't understand the point of this test. `_set_center` is called immediately during 3d data object initialization. Way before any field is involved. Checking the `center` attribute right after object creation should be sufficient:

```
center = ds.domain_center
for test_shape in [(1, 3), (1, 1, 3)]:
    new_center = center.reshape(test_shape)
    assert_equal(ds.sphere(new_center, 0.25).center, center)
    assert_equal(ds.slice(0, 0.25, center=new_center).center, center)
    assert_equal(ds.region(new_center, [-0.25, -0.25, -0.25], [0.25, 0.25, 0.25]).center, center)
```
 BTW, as it is now, it takes almost 5s to run on Travis, which is a lot.